pkg/helper/client.go

```diff
@@ -136,3 +147,34 @@ func ApplyDesiredState(nodeNetworkState *nmstatev1alpha1.NodeNetworkState) (stri
 
 	return setOutput + outputVlanFiltering, nil
 }
+
+func filterOut(currentState nmstatev1alpha1.State, interfacesFilter string) (nmstatev1alpha1.State, error) {
+	if interfacesFilter == "" {
+		return currentState, nil
+	}
+
+	var state map[string]interface{}
+	err := yaml.Unmarshal([]byte(currentState), &state)
+	if err != nil {
+		return currentState, err
+	}
+
+	interfaces := state["interfaces"]
+	var filteredInterfaces []interface{}
+
+	for _, iface := range interfaces.([]interface{}) {
+		name := iface.(map[string]interface{})["name"]
+		g := glob.MustCompile(interfacesFilter)
```

Comment:
maybe move this to the init funtion and save it? we don't need to do it on every call.

The `MustCompile` function comment
```
MustCompile is the same as Compile, except that if Compile returns error, this will panic
```

This will exit I don't think we want that.

Reply:
If I move this `MustCompile` to init function we would couldn't use unit test for testing this feature as it would filter names only from environment variable or configuration file. 
Do you have any better idea how to solve this issue?

Reply:
maybe I don't understand the issue but I think you can create a function for this and you call the function in the init function.

for the unit test just test the function you create that is the same you call from the init